job/job.go

```diff
@@ -232,11 +232,30 @@ func (sg *scopedGroup) Add(jobs ...Job) {
 	sg.group.add(sg.health, jobs...)
 }
 
-var nameRegex = regexp.MustCompile(`^[a-zA-Z][a-zA-Z0-9_\-]{0,100}$`)
-
-func validateName(name string) error {
-	if !nameRegex.MatchString(name) {
-		return fmt.Errorf("invalid job name: %q, expected to match %q", name, nameRegex)
+const maxNameLength = 100
+
+func sanitizeName(name string) string {
+	runes := []rune(name)
+	mangled := false
+	for i, r := range runes[:min(maxNameLength, len(name))] {
+		switch {
+		case r >= 'a' && r <= 'z':
```

Comment:
is there no `unicode.Alphanumeric` or similar?

Reply:
I kind of prefer being very explicit here so it's easy to see what's allowed.